### ISSUE TITLE
Question regarding viewing custom tlv records

### ISSUE THREAD

**Issue by Chris869218** [alexbosworth/balanceofsatoshis#100]
If I was trying to distinguish between the following two scenarios:

1) Keysend invoices coming into my node contain custom tlv records, but I don't know how to view them through BOS.
2) The keysend invoices coming into my node do not contain custom tlv records, and that is why I can't find a way to view them in BOS.

How would I be able to distinguish between those? These wouldn't be chat messages, but podcast episode names and timestamps that correspond to the payment being sent. See the _getTlv portion of .sendSpontaneousPayment at the following link for the information I am expecting to be contained in the payment, and am hoping to view. https://github.com/breez/breezmobile/blob/ac0882f05485fdad82ac86589e8b4a62a78487a8/lib/bloc/podcast_payments/podcast_payments_bloc.dart

I believe this information is contained in the invoice, but I'm not seeing it in BOS (BOS 8.09, lnd 0.12.0 on a raspiblitz).

Thanks in advance.

**Comment by alexbosworth** [comment]
What method are you using to look at invoices? I don't think there is a command to view invoices other than the accounting one

**Comment by Chris869218** [comment]
bos accounting invoices --csv  . Would any custom tlv record be contained in that csv output if present?

**Comment by alexbosworth** [comment]
Ah ok, currently only two TLV records are considered:

- `34349339`: used to mark a "from" public key
- `5482373484`: used to mark a "keysend" payment

https://github.com/alexbosworth/ln-accounting/blob/master/harmony/invoices_as_records.js#L51

**Comment by Chris869218** [comment]
Any chance other TLV records are in the roadmap? With over 200 podcasts, the "podcast:value" specification usage is growing (https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md). With more and more podcasts utilizing keysend payments that contain the podcast name, episode name, and episode timestamp in the TLV records, it would be great to get at that information through BOS accounting invoices. Currently it seems the best way to get at that information is to have Satoshis.Stream host the node and they collect those TLV records and supply a spreadsheet export. It would be nice to just get at that information though my own node though if possible.

**Comment by alexbosworth** [comment]
It should be already showing up as [KeySend] right?

I'm working on a way to handle TLV protocols other than the regular keysend one at the moment but still in progress

**Comment by Chris869218** [comment]
When I use bos accounting invoices --csv, I see the following columns. Amount, Asset (says BTC, which is a little confusing at first glance as the values in the 'amount' column are in sats), Date & Time, Fiat Amount, From ID (which is blank for these podcast player keysend payments), Network ID (which is also blank), Notes (which says [Push Payment]), To ID (also blank), Transaction ID, and type (listed as 'income'). As a side note, I just saw this (https://github.com/satoshisstream/satoshis.stream/blob/main/TLV_registry.md) for a list of TLV protocols Satoshis.Stream is reading in. Thanks for the consideration.

**Comment by alexbosworth** [comment]
Ok [Push Payment] indicates it's a keysend

BTC indicates the currency type not the unit

**Comment by Chris869218** [comment]
Noted. The 'Fiat Amount' column cleared it up real fast. Just made me do a double take for a second when I saw a column of "8 BTC, 120 BTC" etc.

**Comment by alexbosworth** [comment]
Yeah that style comes from the Harmony standard this is based on: https://github.com/picksco/harmony

**Comment by alexbosworth** [comment]
If you have an example invoice from lookupinvoice with all the data I could look at mapping it into custom accounting metadata
